### PR TITLE
fix: track balance of want on bsc

### PR DESCRIPTION
**Pull request by fp-crypto**

### CODE REVIEW

File: contracts/WethToBscStrategy.sol [hunk 1]
```diff
@@ -26,6 +26,8 @@ contract WethToBscStrategy is BaseStrategy {
     address public ethDepositToBsc =
         address(0x13B432914A996b0A48695dF9B2d701edA45FF264);
 
+    uint256 internal _balanceOfWantOnBSC = 0;
```
Comment (poolpitako) [inline contracts/WethToBscStrategy.sol]: underscore is used of read-only, since you are modifying, plz remove the `_`.

Comment (fp-crypto) [reply contracts/WethToBscStrategy.sol]: Is there a good style guide somewhere?

File: contracts/EthereumWethVaultProxy.sol [hunk 2]
```diff
@@ -99,8 +104,12 @@ contract EthereumWethVaultProxy {
         }
     }
 
-    function sendBack() external onlyGovOrStrategist {
-        vault.withdraw();
+    function sendBack(uint256 _sendBackAmount) external onlyGovOrStrategist {
+        assert(estimatedTotalAssets() <= _sendBackAmount);
```
Comment (poolpitako) [inline contracts/EthereumWethVaultProxy.sol]: `require()`

Comment (poolpitako) [reply contracts/EthereumWethVaultProxy.sol]: also, souldn't `estimatedTotalAssets()` be greater than `sendBackAmount`?

File: contracts/EthereumWethStrategyProxy.sol [hunk 3]
```diff
@@ -149,4 +141,92 @@ contract EthereumWethStrategyProxy {
...
+    }
+
+    function estimatedTotalAssets() public view returns (uint256) {
+        return
+            balanceOfEth().add(balanceOfAnyEth()).add(
+                balanceOfVaultSharesInEth()
+            );
+    }
+
+    function migrate(address _newStrategy) external onlyGov {
```
Comment (poolpitako) [inline contracts/EthereumWethStrategyProxy.sol]: `_newProxy` instead of `_newStrategy`? wdyt?

File: contracts/WethToBscStrategy.sol [hunk 4]
```diff
@@ -120,7 +105,16 @@ contract WethToBscStrategy is BaseStrategy {
         return IERC20(want).balanceOf(address(this));
     }
 
-    function prepareMigration(address _newStrategy) internal override {}
+    function proxyDebt() public view returns (uint256) {
```
Comment (poolpitako) [inline contracts/WethToBscStrategy.sol]: remove?

**Review by poolpitako — APPROVED** [review]

File: tests/test_nerve_proxy.py [hunk 5]
```diff
@@ -35,3 +35,73 @@ def test_nerve_proxy(EthereumWethStrategyProxy, owner, strategist):
...
+    proxy.deposit({"from": strategist})
+    proxy_vault_balance = vault.balanceOf(proxy)
+    assert proxy_vault_balance > 0
+
+    any_eth.transfer(proxy, Wei("0.1 ether"), {"from": any_eth_whale})
+    bsc_eth.transfer(proxy, Wei("3 ether"), {"from": bsc_eth_whale})
+    fusdt.transfer(proxy, 17728, {"from": fusdt_whale})
+    assert fusdt.balanceOf(proxy) > 0
+
+    proxy.sweep(fusdt, {"from": owner})
```
Comment (poolpitako) [inline tests/test_nerve_proxy.py:102]: you can test that you can't sweep any_eth by doing:
```
with brownie.reverts():
  proxy.sweep(any_eth, {"from": owner})
```

**Review by poolpitako — APPROVED** [review]

**Pull request merged**